apps/src/blockly/addons/contextMenu.js

```diff
@@ -284,4 +285,16 @@ function isDarkTheme(workspace) {
 function baseName(themeName) {
   return themeName.replace(dark, '');
 }
+
+function unregisterDefaultOptions() {
+  // Remove some default context menu options, if they are present.
+  try {
+    // cleanUp() doesn't currently account for immovable blocks.
+    GoogleBlockly.ContextMenuRegistry.registry.unregister('cleanWorkspace');
+    GoogleBlockly.ContextMenuRegistry.registry.unregister('collapseWorkspace');
+    GoogleBlockly.ContextMenuRegistry.registry.unregister('expandWorkspace');
+    GoogleBlockly.ContextMenuRegistry.registry.unregister('workspaceDelete');
+  } catch (error) {}
```

Comment:
Is this here to catch if a context menu option isn't in the list? Could we just use optional chaining?

Reply:
Ah ok, a couple thoughts then -- the last line of this section is confusing then, seems like that's what it's intended to do?

https://github.com/code-dot-org/code-dot-org/blob/f3b61117bf2ad9de98407b73f23840c015997ff3/apps/test/unit/blockly/googleBlocklyWrapperTest.js#L14-L19

If you can't figure out why that isn't doing what we'd expect/don't want to figure it out now, could you catch a more specific error (maybe not, looks generic from the log you shared)? At the very least, a comment noting why we're catching all errors here that more explicitly notes that it is for tests?

Reply:
So all the code in that `init` method is run multiple times on the same `GoogleBlockly` instance without that instance ever being reset. Resetting the context menu here had been working because previously we were only registering new options to the context menu, not unregistering existing ones from it. I can look at this again a little later on. Thanks!